src/Illuminate/Database/Query/Grammars/PostgresGrammar.php

```diff
@@ -242,4 +255,21 @@ public function compileTruncate(Builder $query)
     {
         return ['truncate '.$this->wrapTable($query->from).' restart identity' => []];
     }
+
+    /**
+     * Concatenate an array of segments, removing empties.
+     *
+     * @param  array   $segments
+     * @return string
+     */
+    protected function concatenate($segments)
```

Comment:
This entire function needs some explanation. It's hard to follow and is unclear what is going on.